SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialActorChannel.cpp

```diff
@@ -1172,11 +1163,33 @@ void USpatialActorChannel::UpdateSpatialPosition()
 			return;
 		}
 	}
+	
+	// Check that the Actor has satisfied both minimum thresholds OR either of the maximum thresholds
+	bool bSatisifiedPositionUpdateRequirements = false;
+
+	const float TimeSinceLastPositionUpdate = NetDriver->GetElapsedTime() - TimeWhenPositionLastUpdated;
 
-	// Check that the Actor has moved sufficiently far to be updated
-	const float SpatialPositionThresholdSquared = FMath::Square(GetDefault<USpatialGDKSettings>()->PositionDistanceThreshold);
 	FVector ActorSpatialPosition = SpatialGDK::GetActorSpatialPosition(Actor);
-	if (FVector::DistSquared(ActorSpatialPosition, LastPositionSinceUpdate) < SpatialPositionThresholdSquared)
+	const float DistanceTravelledSinceLastUpdateSquared = FVector::DistSquared(ActorSpatialPosition, LastPositionSinceUpdate);
+
+	const USpatialGDKSettings* SpatialGDKSettings = GetDefault<USpatialGDKSettings>();
+	const float SpatialMinimumPositionThresholdSquared = FMath::Square(SpatialGDKSettings->PositionUpdateThresholdMinDistance);
+	const float SpatialMaximumPositionThresholdSquared = FMath::Square(SpatialGDKSettings->PositionUpdateThresholdMaxDistance);
+
+	if (TimeSinceLastPositionUpdate >= SpatialGDKSettings->PositionUpdateThresholdMinSeconds && DistanceTravelledSinceLastUpdateSquared >= SpatialMinimumPositionThresholdSquared)
+	{
+		bSatisifiedPositionUpdateRequirements = true;
+	}
+	else if (TimeSinceLastPositionUpdate >= SpatialGDKSettings->PositionUpdateThresholdMaxSeconds && DistanceTravelledSinceLastUpdateSquared > 0.0f)
```

Comment:
Interesting question that might not be worth thinking too hard about - do we want to start timing from when it last moved. So if the movement is 0 do we actually want to reset the timer?
I think earlying out from movement being 0 earlier on would make this moot, but I'd argue that 0 movement means you are still up to date. And you want the timer to represent how long your data has been stale for.

Reply:
I do agree with you on this, but what might be up for consideration is what we consider 0 to actually be? I mean, shall we just accept every movement, as small as it is as something that we need to update for?

Reply:
For now I'd use 0. We could set a true minimum that is considered 0 but I think we can revisit that later with data.

Reply:
Okay, that makes sense, thank you!